### PR TITLE
Refactore

### PR DESCRIPTION
**Pull request by Lockless404**
Refactored App and main files.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+def create_rental
+  puts 'Select a book from the following list by number'
+  @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+  book_id = gets.chomp.to_i
+
+  puts 'Select a person from the following list by number (not id)'
+  @people.each_with_index do |person, index|
+    puts "#{index}) [#{person.classroom}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
```
Comment (TedLivist) [inline app.rb]: - [ ] When I try to list the teacher object for creating a rental, I get an error message about an undefined classroom method for a teach. Kindly fix that. I think what you meant here is `person.class`, not `person.classroom`.

  lockless-teacher

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
```
Comment (TedLivist) [inline app.rb]: - [ ] There seems to be a violation of the DRY and the Single-responsibility rules in the **`create_student`** and **`create_teacher`** methods. Both methods request for **`Age`** and **`Name`** inputs, this breaks the DRY rule. Also, user inputs and input validation should be handled by another in keeping with the Single-responsibility rule of SOLID. In keeping with both rules, kindly refactor your code accordingly.

  **EXAMPLE:** You can extract the requests into a new array and then return age and name in an array from the new method. Next, destructure the requests by calling the new method in **create_student** and **create_teacher**

  New method | Uses of new methods
  :---:|:---:
  lockless-person | lockless-stud-teach

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    sleep 0.75
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
```
Comment (TedLivist) [inline app.rb]: - [ ] In keeping with the Single-responsibility rule, kindly extract input requests and validation to another method.

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  puts 'Select a person from the following list by number (not id)'
+  @people.each_with_index do |person, index|
+    puts "#{index}) [#{person.classroom}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  end
+
+  person_id = gets.chomp.to_i
+
+  print 'Date: '
+  date = gets.chomp.to_s
```
Comment (TedLivist) [inline app.rb]: - [ ] In keeping with the Single-responsibility rule, kindly extract requests for user input and validation to another method.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Lockless404,

Good job so far!
✔️ Good documentation
✔️ Nice refactoring
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
Hi @Lockless404 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**